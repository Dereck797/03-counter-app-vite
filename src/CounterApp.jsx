import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({value}) => { 

    const [counter, setCounter] = useState( value );

    const suma = () => setCounter((co) => co + 1);
    const resta = () => setCounter((co) => co - 1);
    const reset = () => setCounter(value);
    
    return (
        <>
        <h1>CounterApp</h1>
        <h2 data-testid="counter">{ counter }</h2>

        <button onClick={suma} > +1 </button>
        <button onClick={resta} > -1 </button>
        <button onClick={reset} > Reset </button>
        </>
    )
}


CounterApp.propTypes = {
    value: PropTypes.number
}