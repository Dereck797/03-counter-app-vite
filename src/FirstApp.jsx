//los porp types sirven para definir el tipo de las propiedades o props
import PropTypes from 'prop-types';

export const FirstApp = ({title, subtitle }) => {

  return (
    <>
      <h1>{title}</h1>
      <h1>{subtitle}</h1>
    </>
  )
}

// En este caso estamos definiendo que title no puede ser nada mas que un String
// Si nosotros enviamos un número al componente como propiedad a title este presentará 
// un error en consola
FirstApp.propTypes = {
  // La etiqueta isRequired define que la propiedad debe ser obligatoria
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number.isRequired
}
