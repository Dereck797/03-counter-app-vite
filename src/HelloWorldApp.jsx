// export function App(){
//     return <h1>Hola Mundo</h1>
// }

// export const HelloWorldApp = () => {
//   return (
//     <div>HelloWorldApp</div>
//   )
// }

export const HelloWorldApp = () => <h1>HelloWorldApp</h1> ;
