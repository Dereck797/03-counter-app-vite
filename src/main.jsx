import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
import { HelloWorldApp } from './HelloWorldApp';

// Se importan los estilos globales
import './styles.css'

//
ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <FirstApp></FirstApp>
        {/* <CounterApp value={ 234 }/> */}
    </React.StrictMode>
);