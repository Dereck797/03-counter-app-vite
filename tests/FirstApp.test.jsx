import { render } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en FirstApp', () => {

    test('Debe de hacer match con el snapshot', () => {
      
        const title = 'Hola Soy Goku Super Sayayin 4';
        // el render retorna un objeto que expone ciertas propiedades dentro de ellas 
        // tenemos al container
        const { container } = render( <FirstApp title={title}/> )
        console.log( container );
        
        // El método toMatchSnapshot crea una fotografía del objeto renderizado inicialmente
        // Automáticamente se crea un carpeta llamada snapshots en donde se ve reflejado el snapshot inicial 
        // No tiene mucho sentido utilizar este método para componentes que cambian mucho
        expect( container ).toMatchSnapshot();

    })

    test('Debe de mostrar el titulo H1', () => {
      
        const title = 'Hola Soy Goku Super Sayayin 4';
        // Estas funciones que nos da testing Library son muy utiles para poder anailzar rápidamente el DOM(Es el componente renderizado)
        const { container, getByText } = render( <FirstApp title={title}/> )
        // esta función nos permite saber si en lo que estamos renderizando existe ese titulo
        expect( getByText(title) ).toBeTruthy();

        // El querySelector me permite saltar al DOM y buscar una etiqueta H1
        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(title);

    })
    
    
  
})
