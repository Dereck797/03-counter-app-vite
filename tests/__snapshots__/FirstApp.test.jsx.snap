// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola Soy Goku Super Sayayin 4
  </h1>
  <h1>
    No Hay Subtitulo
  </h1>
  <h1>
    Dereck Mu√±oz
  </h1>
</div>
`;
